### DISCUSSION TITLE
Is it possible to just have premake produce a batch file?

### DISCUSSION THREAD

**Discussion by bogez57** [premake/premake-core#2016] [Q&A] [answered]
I don't use visual studio or any other heavy duty IDE. I generally just write batch files to build projects. However, it would be nice to have a more readable, manageable way to write these files. Is premake able to generate just a batch file which contains all the compiler switches and such?

**Comment by nickclark2016** [comment, marked as answer]
In the core repository, we have no such functionality.  It's possible that there exists a 3rd party module to do exactly that.  For VS specifically, you can still generate the solution and project files, then use `msbuild` to compile, rather than using the full VS IDE (assuming you're using the VS exporter).

**Reply by bogez57** [reply]
Okay, thanks for the information.

**Comment by Jarod42** [comment]
For non-IDE generators,
There is builtin gmake/gmake2 which generate makefile (gcc/clang).
As external [build-system Modules](https://premake.github.io/community/modules/#build-system-modules), there is also premake-ninja for ninja (gcc/clang/msvc) or premake-cmake for CMake(gcc/clang/msvc).
